Sockets/BackEndMonitoringSockets/include/CSocket.h

```diff
@@ -4,17 +4,19 @@
 class CSocket : public CBaseSocket
 {
 public:
-	CSocket() = delete;
+	CSocket() = default;
 	explicit CSocket(const int port, const std::string& ip_address);
-	CSocket(const CSocket&) = delete;
-	CSocket(CSocket&&) noexcept = delete;
-	~CSocket() noexcept = default;
+	explicit CSocket(const int port, const std::string& ip_address, int socket_fd);
+	explicit CSocket(sockaddress address, int socket_fd);
+	CSocket(const CSocket& obj) = default;
+	CSocket(CSocket&& obj) noexcept = default;
+	CSocket& operator = (const CSocket& obj) = default;
 
-	bool CloseSocket();
 	[[nodiscard]]  sockaddress GetSocketAddress() const;
-
-protected:
-	void SetSocketAddress(const int port, const std::string& ip_address);
+	void InitAddress(const int port, const std::string& ip_address);
 	bool IsValidSocket() const;
-	
+
+private:
+	sockaddress m_address;
```

Comment:
I don't that CSocket, CSocketWrapper, CBaseSocket and all other socket classes should not store anything except socket descriptor.

Reply:
Fixed, I add class CSockAddress which hold the address and can get where it is required